### PR TITLE
feat: support header sync feature

### PR DESCRIPTION
**Pull request by driftluo**

### CODE REVIEW

File: src/rpc_server.rs [hunk 1]
```diff
@@ -127,19 +136,42 @@ impl EmitterServer for EmitterRpc {
...
+                IndexerTip {
+                    block_hash: new_header.hash,
+                    block_number: new_header.inner.number,
+                }
+            };
+            let raw = self
+                .state
+                .header_state
+                .0
+                 .0
```
Comment (felicityin) [inline src/rpc_server.rs:168]: extra space

File: src/global_state.rs [hunk 2]
```diff
@@ -51,27 +120,82 @@ impl GlobalState {
...
+                        let header = client.get_header_by_number(i.into()).await.unwrap();
+                        headers.push(header);
+                    }
 
-                Arc::new(cells.into_iter().collect())
+                    submit_headers(headers).await;
+
+                    let raw = state
+                        .0
+                         .0
```
Comment (felicityin) [inline src/global_state.rs:166]: extra space

**Review by felicityin — APPROVED** [review]

File: src/rpc_server.rs (hunk 1 above)
Comment (driftluo) [reply src/rpc_server.rs:168]: It seems that the default formatting is like this and is not changed manually

Comment (felicityin) [reply src/rpc_server.rs:168]: ok

**Pull request merged**